lib/src/components/NxPageHeader/__tests__/NxPageHeader.test.tsx

```diff
@@ -5,84 +5,117 @@
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
 import React from 'react';
-import { shallow } from 'enzyme';
-import 'jest-enzyme';
-
-import * as enzymeUtils from '../../../__testutils__/enzymeUtils';
 import NxPageHeader from '../NxPageHeader';
-import AbstractNxPageHeader from '../../AbstractNxPageHeader/AbstractNxPageHeader';
+
+import { screen } from '@testing-library/react';
+import { rtlRenderElement } from '../../../__testutils__/rtlUtils';
 
 describe('NxPageHeader', function() {
-  const getShallowComponent = enzymeUtils.getShallowComponent(NxPageHeader, {});
 
-  it('renders an AbstractNxPageHeader', function() {
-    expect(getShallowComponent()).toMatchSelector(AbstractNxPageHeader);
+  const renderEl = rtlRenderElement(NxPageHeader, {});
+
+ it('renders a top-level header element', function() {
+    renderEl();
+    const header = screen.getByRole('banner');
+    expect(header).toBeInTheDocument();
   });
 
-  it('passes its links, homeLink, and children props to the AbstractNxPageHeader', function() {
-    const props = {
-          links: [{ name: 'foo', href: '#bar' }],
-          homeLink: '#home',
-          children: <button>Click Here For a Free iPhone</button>
-        },
-        component = getShallowComponent(props);
+  it('renders the default logo within the header', function() {
+    renderEl();
+    const logo = screen.getByRole('img');
```

Comment:
To make sure that the image is within the header, as the test name says, what you really want to to is use RTL queries based off of the `banner` that the component renders, not off of the entire `screen`.  We already verified in the test above that the top-level element is the `banner`, so that means you can query off of that top-level element.  This is an extremely common use-case in RSC's RTL tests, and so we have a utility to help with it, which, to be consistent, you should really be using most everywhere that you're currently using `screen`.  That utility is the `rtlRender` function in `rtlUtils`.  It works similarly to `rtlRenderElement`: you pass it the component itself and the minimal props, and it returns a function (which we typically call `quickRender`).  That function can optionally take more props, and it renders the element to the page.  The difference is in what it returns. While `renderEl` returns the top-level element, `quickRender` returns an RTL `RenderResult` that's been modified to have all of its query functions bound to the rendered element rather than `screen`.  So using that infrastructure, the way I'd write this test would be something like this:
```
const view = quickRender(),
    logo = view.getByRole('img');

expect(logo).toBeInTheDocument();
```

Reply:
> To make sure that the image is within the header, as the test name says, what you really want to do is use RTL queries based off of the `banner` that the component renders, not off of the entire `screen`. We already verified in the test above that the top-level element is the `banner`, so that means you can query off of that top-level element. This is an extremely common use-case in RSC's RTL tests, and so we have a utility to help with it, which, to be consistent, you should really be using most everywhere that you're currently using `screen`. That utility is the `rtlRender` function in `rtlUtils`. It works similarly to `rtlRenderElement`: you pass it the component itself and the minimal props, and it returns a function (which we typically call `quickRender`). That function can optionally take more props, and it renders the element to the page. The difference is in what it returns. While `renderEl` returns the top-level element, `quickRender` returns an RTL `RenderResult` that's been modified to have all of its query functions bound to the rendered element rather than `screen`. So using that infrastructure, the way I'd write this test would be something like this:
> 
> ```
> const view = quickRender(),
>     logo = view.getByRole('img');
> 
> expect(logo).toBeInTheDocument();
> ```

Sounds good to me. I refactored the code here: [67d3ddb](https://github.com/sonatype/sonatype-react-shared-components/pull/935/commits/67d3ddb6e80eea79e525764fca3eb48f940c9ff2)